tests/Language/Java/InlineSpec.hs

```diff
@@ -52,6 +56,11 @@ spec = do
           _ :: JJObject <- [java| $obj1 |]
           return ()
 
+        it "Supports parameterized type synonyms" $ do
+          obj :: List ObjectClass <- [java| new Object() {} |]
```

Comment:
Shouldn't the java qq return a list as the type prescribes? And perhaps more important, shouldn't we have a runtime casting error?